static/js/components/email/EmailCompositionDialog.js

```diff
@@ -28,25 +31,49 @@ type EmailDialogProps = {
   email:            EmailState,
   sendEmail:        () => void,
   title?:           string,
-  children?:        any,
+  subheadingType?:  string
+};
+
+const renderSimpleSubheading = (subheading: string) => (
+  <div className="subheading-section">
+    <h5 className="subheading">{ subheading }</h5>
+  </div>
+);
+
+const renderRoundedSubheading = (subheading: string) => (
+  <div className="subheading-section">
+    <Grid noSpacing={true}>
+      <Cell col={1} align={"middle"} className="subheading-recipient">TO:</Cell>
+      <Cell col={11}><h5 className="subheading rounded">{ subheading }</h5></Cell>
+    </Grid>
+  </div>
+);
+
+const emailTypeRendererMap = {
+  [SEARCH_EMAIL_TYPE]: renderSimpleSubheading,
+  [COURSE_EMAIL_TYPE]: renderRoundedSubheading
+};
```

Comment:
It might be nice to use a `Map` here, then down below where you're pulling the render function out of here you could just do:

```js
let renderFunc = emailTypeRenderedMap.get(renderType) || renderSimpleSubheading;
```

(taking advantage of that fact that if you call `.get` with a key the map in question does not contain it returns `undefined`)